src/figuresClasses.ts

```diff
@@ -2,10 +1,109 @@
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  color;
+
+  constructor(
+    color: string,
+
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    this.color = color;
+
+    if (a <= 0) {
+      throw new Error('`a` can`t be lower or equal zero');
+    }
 
-export class Triangle implements Figure {}
+    if (b <= 0) {
+      throw new Error('`b` can`t be lower or equal zero');
+    }
+
+    if (c <= 0) {
+      throw new Error('`c` can`t be lower or equal zero');
+    }
+
+    const longest: number = Math.max(a, b, c);
+
+    if (
+      (a === longest && a >= b + c) ||
+      (b === longest && b >= a + c) ||
+      (c === longest && c >= b + a)
+    ) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
+
+    return (
+      Math.floor(
+        Math.sqrt(
+          semiPerimeter *
+            (semiPerimeter - this.a) *
+            (semiPerimeter - this.b) *
+            (semiPerimeter - this.c),
+        ) * 100,
+      ) / 100
+    );
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  shape = 'circle';
 
-export class Rectangle implements Figure {}
+  color;
+
+  constructor(
+    color: string,
+    private radius: number,
+  ) {
+    this.color = color;
+
+    if (radius <= 0) {
+      throw new Error('`radius` can`t be lower or equal zero');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor(Math.PI * Math.pow(this.radius, 2) * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape = 'rectangle';
+
+  color;
```

Comment:
The `color` property is declared but not initialized. Consider initializing it directly in the declaration or within the constructor.